### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by celinevo**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -69,8 +32,8 @@ export default function App() {
           element={
             <CreepypastaPage
               handleBookmarkClick={handleBookmarkClick}
```
Comment (Quaquaro) [inline src/App.js:34]: Since you are passing the function, it would be correct to use the prefix "on" after convention.

File: src/App.js [hunk 2]
```diff
@@ -69,8 +32,8 @@ export default function App() {
...
-              creepypastas={creepypastas}
               searchValue={searchValue}
+              creepypastas={creepypastas}
               handleChange={handleChange}
```
Comment (Quaquaro) [inline src/App.js]: Since you are passing the function, it would be correct to use the prefix "on" after convention.

File: src/App.js [hunk 3]
```diff
@@ -85,17 +48,8 @@ export default function App() {
...
-              nameEditingValue={nameEditingValue}
-              infoEditingValue={infoEditingValue}
-              onNameChange={onNameChange}
-              onInfoChange={onInfoChange}
-              onNameBlur={onNameBlur}
-              onInfoBlur={onInfoBlur}
-              onKeyDown={onKeyDown}
-              profileImage={profileImage}
-              profileUpload={profileUpload}
               handleBookmarkClick={handleBookmarkClick}
```
Comment (Quaquaro) [inline src/App.js]: Since you are passing the function, it would be correct to use the prefix "on" after convention.

File: src/App.js [hunk 4]
```diff
@@ -85,17 +48,8 @@ export default function App() {
...
+              creepypastas={creepypastas}
               handleDeleteCreepypasta={handleDeleteCreepypasta}
```
Comment (Quaquaro) [inline src/App.js]: Since you are passing the function, it would be correct to use the prefix "on" after convention.

File: src/App.js [hunk 5]
```diff
@@ -85,17 +48,8 @@ export default function App() {
...
               handleRedirectEdit={handleRedirectEdit}
```
Comment (Quaquaro) [inline src/App.js]: Since you are passing the function, it would be correct to use the prefix "on" after convention.

File: src/App.js [hunk 6]
```diff
@@ -114,15 +68,6 @@ export default function App() {
               )}
               handleBookmarkClick={handleBookmarkClick}
```
Comment (Quaquaro) [inline src/App.js]: Since you are passing the function, it would be correct to use the prefix "on" after convention.

File: src/App.js [hunk 7]
```diff
@@ -152,6 +97,7 @@ export default function App() {
     </AppGrid>
   );
```
Comment (Quaquaro) [inline src/App.js]: This also applies to the other passed functions

File: src/App.js [hunk 8]
```diff
@@ -190,27 +140,6 @@ export default function App() {
...
-      .then(onProfileImageSave)
-      .catch(err => console.error(err));
-  }
-
-  function onProfileImageSave(response) {
-    setProfileImage(response.data.url);
-  }
 }
 
 const AppGrid = styled.div`
```
Comment (Quaquaro) [inline src/App.js:135]: Confusing name, as you don't have a grid in it at all

File: src/components/BookmarkButton.js [hunk 9]
```diff
@@ -1,15 +1,15 @@
...
     <Bookmark type="button" onClick={onBookmarkClick}>
       {isBookmarked ? (
-        <img src={BookmarkIconActive} alt="Story is spookmarked" width="35px" />
+        <img src={bookmarkIconActive} alt="Story is spookmarked" width="35px" />
       ) : (
         <img
-          src={BookmarkIconInactive}
+          src={bookmarkIconInactive}
           alt="Story is not spookmarked"
           width="35px"
```
Comment (Quaquaro) [inline src/components/BookmarkButton.js:14]: Add also a height here.

File: src/components/BookmarkButton.js [hunk 10]
```diff
@@ -1,15 +1,15 @@
...
-import BookmarkIconInactive from '../Images/grayskull.svg';
+import bookmarkIconActive from '../Images/redskull.svg';
+import bookmarkIconInactive from '../Images/grayskull.svg';
 
 export default function BookmarkButton({ isBookmarked, onBookmarkClick }) {
   return (
     <Bookmark type="button" onClick={onBookmarkClick}>
       {isBookmarked ? (
-        <img src={BookmarkIconActive} alt="Story is spookmarked" width="35px" />
+        <img src={bookmarkIconActive} alt="Story is spookmarked" width="35px" />
```
Comment (Quaquaro) [inline src/components/BookmarkButton.js:9]: Add also a height here.

File: src/components/CustomCard.js [hunk 11]
```diff
@@ -45,10 +45,10 @@ export default function CustomCard({
             </RedButton>
           </ReadMorePosition>
           <EditIcon onClick={() => handleRedirectEdit(id)}>
-            <img src={PenIcon} alt="Edit this creepypasta" width="35px" />
+            <img src={penIcon} alt="Edit this creepypasta" width="35px" />
```
Comment (Quaquaro) [inline src/components/CustomCard.js:48]: Add also a height here.

File: src/components/CustomCard.js [hunk 12]
```diff
@@ -45,10 +45,10 @@ export default function CustomCard({
...
           </EditIcon>
           <DeleteIcon onClick={() => setShowMessage(true)}>
-            <img src={TrashIcon} alt="Delete this creepypasta" width="35px" />
+            <img src={trashIcon} alt="Delete this creepypasta" width="35px" />
```
Comment (Quaquaro) [inline src/components/CustomCard.js:51]: Add also a height here.

File: src/components/Form.js [hunk 13]
```diff
@@ -129,6 +129,12 @@ const Label = styled.label``;
 const Input = styled.input`
   border-radius: 6px;
   padding-bottom: 0px;
+  border: 2px solid var(--passive);
+  cursor: pointer;
+  &:focus {
+    outline: none;
+    border: 2.5px solid var(--active-main);
```
Comment (Quaquaro) [inline src/components/Form.js:136]: You should better use only whole pixels

File: src/components/Form.js [hunk 14]
```diff
@@ -140,6 +146,12 @@ const Textarea = styled.textarea`
   height: 150px;
   padding: 2px;
   border-radius: 6px;
+  border: 2px solid var(--passive);
+  cursor: pointer;
+  &:focus {
+    outline: none;
+    border: 2.5px solid var(--active-main);
```
Comment (Quaquaro) [inline src/components/Form.js:153]: You should better use only whole pixels

File: src/components/Form.js [hunk 15]
```diff
@@ -129,6 +129,12 @@ const Label = styled.label``;
 const Input = styled.input`
   border-radius: 6px;
   padding-bottom: 0px;
+  border: 2px solid var(--passive);
+  cursor: pointer;
+  &:focus {
+    outline: none;
+    border: 2.5px solid var(--active-main);
+  }
```
Comment (Quaquaro) [inline src/components/Form.js:137]: Also remember the on prefix for the handleCreateCreepypasta and onCreepypastaEdit.

**Review by ttaannjjaa — APPROVED** [review]
looks all very good to me 😃👍

File: src/components/FullCreepypasta.js [hunk 16]
```diff
@@ -1,5 +1,5 @@
-import styled from 'styled-components';
 import { Link } from 'react-router-dom';
+import styled from 'styled-components';
```
Comment (Quaquaro) [inline src/components/FullCreepypasta.js:2]: Remember to give the img-tags a height here as well.

File: src/components/Navigation.js [hunk 17]
```diff
@@ -1,5 +1,5 @@
-import styled from 'styled-components';
 import { NavLink } from 'react-router-dom';
+import styled from 'styled-components';
```
Comment (Quaquaro) [inline src/components/Navigation.js:2]: Remember to give the img-tags a width here as well.

File: src/components/Profile.js [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+  const [profileImage, setProfileImage] = useLocalStorage('ProfileImage', '');
+  const [nameValue, setNameValue] = useLocalStorage('NameKey', 'Your name');
+  const [nameEditingValue, setNameEditingValue] = useState(nameValue);
+  const [infoValue, setInfoValue] = useLocalStorage(
+    'InfoKey',
+    'Here you can write something about yourself!'
+  );
+  const [infoEditingValue, setInfoEditingValue] = useState(infoValue);
+
+  const onNameChange = event => setNameEditingValue(event.target.value);
```
Comment (Quaquaro) [inline src/components/Profile.js:23]: You don't need to put an on in front of the definition.

File: src/components/Profile.js [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+    }
+  };
+
+  return (
+    <>
+      <Position>
+        <PictureCropper>
+          <ProfilePicture
+            alt="Person that created this profile"
+            height="150px"
```
Comment (Quaquaro) [inline src/components/Profile.js:54]: Set a width attribute here.

File: src/components/Profile.js [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+          onChange={onNameChange}
+          onKeyDown={onKeyDown}
+          value={nameEditingValue}
+          onBlur={onNameBlur}
+          maxLength="10"
+        />
+        <PenIcon
+          src={editPen}
+          alt="You can edit the name next to the pen"
+          height="25px"
```
Comment (Quaquaro) [inline src/components/Profile.js:70]: Set a width attribute here.

File: src/components/Profile.js [hunk 21]
```diff
@@ -0,0 +1,206 @@
...
+        />
+      </Position>
+      <AddImage>
+        <label htmlFor="file-upload">
+          <img
+            src={addIcon}
+            alt="Press here to add a profilepicture"
```
Comment (Quaquaro) [inline src/components/Profile.js:77]: Set a width attribute here.

File: src/components/Profile.js [hunk 22]
```diff
@@ -0,0 +1,206 @@
...
+          name="file"
+          onChange={profileUpload}
+          hidden
+        />
+      </AddImage>
+      <ProfileInfo>
+        About me
+        <NavLink to="/feelgood">
+          <img
+            width="45px"
```
Comment (Quaquaro) [inline src/components/Profile.js:93]: Set a height attribute here.

File: src/components/Profile.js [hunk 23]
```diff
@@ -0,0 +1,206 @@
...
+`;
+
+const AddImage = styled.div`
+  position: absolute;
+  left: 98px;
+  top: 90px;
+`;
+
+const ProfileInfo = styled.h2`
+  margin: 15px 0px -20px 0px;
```
Comment (Quaquaro) [inline src/components/Profile.js]: You do not need the last 0

File: src/components/Profile.js [hunk 24]
```diff
@@ -0,0 +1,206 @@
...
+import sparkles from '../Images/sparkles.svg';
+import editPen from '../Images/pen_darkgray.svg';
+import addIcon from '../Images/add.svg';
+
+const CLOUDNAME = process.env.REACT_APP_CLOUDINARY_CLOUDNAME;
+const PRESET = process.env.REACT_APP_CLOUDINARY_PRESET;
+
+export default function Profile() {
+  const [profileImage, setProfileImage] = useLocalStorage('ProfileImage', '');
+  const [nameValue, setNameValue] = useLocalStorage('NameKey', 'Your name');
```
Comment (Quaquaro) [inline src/components/Profile.js:15]: This should also work with only one LocalStorage entry

File: src/components/Searchbar.js [hunk 25]
```diff
@@ -28,6 +28,12 @@ const SearchForm = styled.form`
 const InputField = styled.input`
   border-radius: 6px;
   padding-bottom: 0px;
+  border: 2px solid var(--passive);
+  cursor: pointer;
+  &:focus {
+    outline: none;
+    border: 2.5px solid var(--active-main);
```
Comment (Quaquaro) [inline src/components/Searchbar.js]: Avoid using half pixels

File: src/pages/EditCreepypasta.js [hunk 26]
```diff
@@ -11,7 +11,7 @@ export default function CreatePage({ onEditCreepypasta, creepypastaEdit }) {
       <Flex>
         <Wrapper>
           <GoBackButton onClick={() => navigate(-1)}>
-            <img src={Arrow} alt="Go back" width="60" />
+            <img src={arrow} alt="Go back" width="60" />
```
Comment (Quaquaro) [inline src/pages/EditCreepypasta.js:14]: Add a height here

**Review by Quaquaro — COMMENTED** [review]
Hey Celine, your code is already very well structured, as far as I can tell. I have left you a few suggestions in the code.

**Pull request merged**